### ISSUE TITLE
How does nvm.fish 2.0 compare to Schniz/fnm?

### ISSUE THREAD

**Issue by franciscolourenco** [jorgebucaran/nvm.fish#82]
Hi @jorgebucaran do you have thoughts on https://github.com/Schniz/fnm ?
Would you recommend using `fish-nvm` instead?

Thanks!

**Comment by jorgebucaran** [comment]
I'll try to be fair in this comparison, but it might be difficult. You've been warned. Let's dive in!

- ~~_fnm_ appends code to your `config.fish` (or creates one if there isn't one). I didn't expect this.~~
- _fnm_ is written in ~~OCaml~~ Rust, which is super interesting in its own right.
  - **nvm.fish** is written in Fish, which is the most logical choice for a shell utility that demands seamless shell integration.  
- _fnm_ pollutes my home directory, it saves its state in `~/.fnm` 
  - **nvm.fish** follows the [XDG Base Directory](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) spec just like Fish does.
- _fnm_ requires you to launch a new shell after installing it
  - **nvm.fish** works out of the box via Fisher.
- _fnm_ runs `fnm env --multi | source` on shell startup. I am not a fan of piping stuff to `source`.  
  - **nvm.fish** uses Fish canonical autorun mechanism: configuration snippets.
- I had to curl the _fnm_ installer and pipe it to `bash` to install _fnm_.
  - **nvm.fish** is easier to install, update, and uninstall via Fisher.
- I had to debug an error after installing _fnm_. It turns out I had to run `brew install openssl`. 
  - **nvm.fish** works out of the box. No dependencies other than Fish are required.
- _fnm_ completions are mediocre. 
  - **nvm.fish** tab completions are amazing.
    - `nvm install`<kbd>Tab</kbd> shows you every available version to install + aliases. ([Screen](https://user-images.githubusercontent.com/56996/101286397-a3ef8800-382d-11eb-9596-f30413704f99.jpg))
    - `nvm use`<kbd>Tab</kbd> shows you only versions you can use (already downloaded) + aliases. ([Screen](https://user-images.githubusercontent.com/56996/101286393-9e923d80-382d-11eb-9c04-30023358b050.jpg))
    - `nvm uninstall`<kbd>Tab</kbd> shows you only versions you can uninstall + aliases. ([Screen](https://user-images.githubusercontent.com/56996/101286396-a2be5b00-382d-11eb-9520-6fdfe3d8ed24.jpg))

## Benchmarks

<a id="bench"></a>

```console
$ ~/D/h/bin> time fish -c "fnm install latest"
Looking for node latest for darwin x64
Downloading https://nodejs.org/dist/latest/node-v11.14.0-darwin-x64.tar.xz to /Users/jb/.fnm/downloads/v11.14.0.tar.xz
Extracting /Users/jb/.fnm/downloads/v11.14.0.tar.xz to /Users/jb/.fnm/node-versions/v11.14.0
        9.08 real         1.49 user         1.70 sys
$ ~/D/h/bin> time fish -c "nvm use latest"
fetching https://nodejs.org/dist/v11.14.0/node-v11.14.0-darwin-x64
######################################################################## 100.0%
        8.05 real         1.02 user         1.44 sys
```

**Comment by franciscolourenco** [comment]
Thanks!

**Comment by jorgebucaran** [comment]
https://github.com/Schniz/fnm/issues/26

**Comment by jorgebucaran** [comment]
I just released [2.0](https://github.com/jorgebucaran/nvm.fish/releases/tag/2.0.0). The new **nvm.fish** has been rewritten for Fish 3.0, and special attention has been paid to performance and optimizing every aspect of it. As a result **nvm.fish** feels snappier than ever while still giving you that seamless integration with Fish.
